File: native-quarkus-api/src/main/java/com/xavierbouclet/nativeapi/ActorService.java
package com.xavierbouclet.nativeapi;

import java.util.*;
import javax.inject.Inject;
import javax.inject.Singleton;
import javax.transaction.Transactional;

@Singleton
@Transactional
class ActorService {

    private ActorRepository repository;
    private MovieRepository movieRepository;

    @Inject
    public ActorService(
            ActorRepository repository,
            MovieRepository movieRepository) {
        this.repository = repository;
        this.movieRepository = movieRepository;
    }

    public List<Actor> list() {
        return repository.findAll().list();
    }


    public Actor add(Actor actor) {
        repository.persist(actor);
        return actor;
    }

    public Actor modify(Actor actor) {
        repository.persist(actor);
        return actor;
    }

    public UUID delete(UUID id) {
        repository.deleteById(id);
        return id;
    }

    public Actor get(UUID id) {
        return repository.findById(id);
    }

    public Actor addMovieToActor(UUID id, UUID idMovie) {
        var movie = movieRepository.findById(idMovie);
        var actor = repository.findById(id);
        movie.actors.add(actor);
        actor.movies.add(movie);
        return modify(actor);
    }

    public Actor removeMovieFromActor(UUID id, UUID idMovie) {
        var movie = movieRepository.findById(idMovie);
        var actor = repository.findById(id);
        movie.actors.remove(actor);
        actor.movies.remove(movie);
        return modify(actor);
    }

}

File: native-quarkus-api/src/main/kotlin/com/xavierbouclet/nativeapi/ActorService.kt
package com.xavierbouclet.nativeapi

import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ActorService(
    @Inject val repository: ActorRepository,
    @Inject val movieRepository: MovieRepository
) {

    fun list(): Iterable<Actor> {

        return repository.findAll().list()
    }

    fun add(actor: Actor): Actor {
        repository.persist(actor)
        return actor
    }

    fun modify(actor: Actor): Actor {
        repository.persist(actor)
        return actor
    }

    fun delete(id: UUID): UUID {
        repository.deleteById(id)
        return id
    }

    fun get(id: UUID): Actor? {
        return repository.findById(id)
    }

    fun addMovieToActor(id: UUID, idMovie: UUID): Actor {
        val movie = movieRepository.findById(idMovie)
        val actor = repository.findById(id)
        movie.actors?.add(actor)
        actor.movies?.add(movie)
        repository.persist(actor)
        return actor
    }

    fun removeMovieFromActor(id: UUID, idMovie: UUID): Actor {
        val movie = movieRepository.findById(idMovie)
        val actor = repository.findById(id)
        movie.actors?.remove(actor)
        actor.movies?.remove(movie)
        repository.persist(actor)
        return actor
    }

}
